app/components/channel_bookmarks/channel_bookmarks.tsx

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo, useState} from 'react';
+import {FlatList, View, type ListRenderItemInfo, type NativeSyntheticEvent, type NativeScrollEvent} from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+import Animated from 'react-native-reanimated';
+
+import {GalleryInit} from '@context/gallery';
+import {useTheme} from '@context/theme';
+import {useChannelBookmarkFiles} from '@hooks/files';
+import ChannelBookmarkModel from '@typings/database/models/servers/channel_bookmark';
+import {fileToGalleryItem, openGalleryAtIndex} from '@utils/gallery';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import AddBookmark from './add_bookmark';
+import ChannelBookmark from './channel_bookmark';
+
+type Props = {
+    bookmarks: ChannelBookmarkModel[];
+    canAddBookmarks: boolean;
+    canDeleteBookmarks: boolean;
+    canDownloadFiles: boolean;
+    canEditBookmarks: boolean;
+    canUploadFiles: boolean;
+    channelId: string;
+    currentUserId: string;
+    publicLinkEnabled: boolean;
+    showInInfo: boolean;
+    separator?: boolean;
+}
+
+const GRADIENT_LOCATIONS = [0, 0.64, 1];
+const SCROLL_OFFSET = 10;
+
+const isCloseToBottom = ({layoutMeasurement, contentOffset, contentSize}: NativeScrollEvent) => {
+    return layoutMeasurement.width + contentOffset.x <= contentSize.width - SCROLL_OFFSET;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        flexDirection: 'row',
+        paddingBottom: 2,
+    },
+    separator: {
+        height: 1,
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        marginTop: 8,
+        marginBottom: 24,
+    },
+    emptyItemSeparator: {
+        width: 16,
+    },
+    addContainer: {
+        width: 40,
+        alignContent: 'center',
+    },
+    gradient: {
+        height: 48,
+        width: 76,
+        position: 'absolute',
+        right: 0,
+    },
+    channelView: {
+        paddingLeft: 8,
```

Comment:
can we also add paddingRight: 8 here? You'll have to probably also reduce the width of the gradient by 8